### PR TITLE
Drop support for Java 7; add support for Java 9-11

### PR DESCRIPTION
**Pull request by davidxia**
Upgrade plugins to be compatible with Java 11.
Fix javadocs that started to throw errors with upgraded plugins.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -133,7 +133,7 @@
     <dependency>
       <groupId>com.spotify</groupId>
       <artifactId>docker-client</artifactId>
-      <version>8.11.4</version>
+      <version>8.14.1-SNAPSHOT</version>
```
Comment (davidxia) [inline pom.xml]: Requires this [docker-client PR](https://github.com/spotify/docker-client/pull/1093) to be merged and released.

**Comment by davidxia** [comment]
cc @spotify/docker-admins

File: pom.xml (hunk 1 above)
Comment (mattnworb) [reply pom.xml]: whats the relation of that PR to this one?

File: .travis.yml [hunk 2]
```diff
@@ -5,8 +5,18 @@ services:
...
+  include:
+    - jdk: openjdk8
+    - jdk: openjdk9
+      before_install:
+        - rm "${JAVA_HOME}/lib/security/cacerts"
+        - ln -s /etc/ssl/certs/java/cacerts "${JAVA_HOME}/lib/security/cacerts"
+    - jdk: openjdk10
+      before_install:
+        - rm "${JAVA_HOME}/lib/security/cacerts"
+        - ln -s /etc/ssl/certs/java/cacerts "${JAVA_HOME}/lib/security/cacerts"
```
Comment (mattnworb) [inline .travis.yml]: whats this about?

File: pom.xml (hunk 1 above)
Comment (mattnworb) [reply pom.xml]: I think you can change this to use 8.14.0 to uncouple the two chanes

File: .travis.yml (hunk 2 above)
Comment (davidxia) [reply .travis.yml]: Ah, it's in my second commit's message.

File: pom.xml (hunk 1 above)
Comment (davidxia) [reply pom.xml]: I was testing docker-maven-plugin locally and ran into a docker-client bug there.

File: .travis.yml (hunk 2 above)
Comment (davidxia) [reply .travis.yml]: I only needed this to be able to use SNAPSHOT deps. I'll remove once docker-client is released.

Comment (mattnworb) [reply .travis.yml]: ah sorry I missed that commit message. It must be Monday

Comment (davidxia) [reply .travis.yml]: Oops, I was also mistaken. These certs need to be mounted regardless of whether you use SNAPSHOT deps or not from sonatype. It's bc openjdk9 and 10 lack these certs...

**Pull request merged**